src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
The lack of nearby collisions were perfect for the pathfinder because it scans locally. but I don't think the old hash function is good for the world as a whole; A worst case thought experiment: players can easily organize an "attack" by standing 128 chunks apart. If you can get enough of them to do this you are guaranteed to get so many collisions you're effectively now worse than a linked list. Note that the collisions will occur not just with the chunk where the players are standing, but in all loaded radius around them.